crates/runtime/src/table.rs

```diff
@@ -57,6 +57,17 @@ impl Table {
         }
     }
 
+    /// Returns the type of the elements in this table.
+    pub fn element_type(&self) -> TableElementType {
+        match &*self.elements.borrow() {
+            TableElements::FuncRefs(_) => TableElementType::Func,
+            #[cfg(target_pointer_width = "32")]
+            TableElements::ExternRefs(_) => TableElementType::Val(ir::types::R32),
+            #[cfg(target_pointer_width = "64")]
+            TableElements::ExternRefs(_) => TableElementType::Val(ir::types::R64),
```

Comment:
FWIW this pattern with `#[cfg]` might be good to encapsulate somewhere with something like `ir::types::rsize()`